presto-main/src/main/java/com/facebook/presto/sql/analyzer/StatementAnalyzer.java

```diff
@@ -561,22 +562,29 @@ protected Scope visitCall(Call node, Optional<Scope> scope)
             return createAndAssignScope(node, scope);
         }
 
-        private void validateColumns(Statement node, RelationType descriptor)
+        private void validateColumns(Statement node, RelationType descriptor, Optional<List<String>> columnAliases)
```

Comment:
I don't like how this method interleaves the validation of the aliases with the validation of the outputs of the subquery. In particular, things such as checking the type of the expression have nothing to do with validating the alias names. Handling missing names is also something that only applies to output columns from the subquery.

Once you make the adjustment I suggested elsewhere regarding using Identifier as way to represent the column name, this code will need to be changed accordingly and may force a cleaner separation.